### PR TITLE
Angular 17 to 18 update

### PR DESCRIPTION
**Pull request by mimiflynn**

### CODE REVIEW

**Comment by mimiflynn** [comment]
THIS SOFTWARE IS CONTRIBUTED SUBJECT TO THE TERMS OF THE FINOS Corporate Contributor License Agreement.

THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.

**Comment by DovOps** [comment]
Thanks Mimi.  This compiles but does not build due to ag-grid compatibility isues with angular 18.  I'm moving the ag-grid changes in here (originally part of PR #207  ).

File: web-front-end/angular/package.json
Comment (DovOps) [inline web-front-end/angular/package.json:1]: ngx-bootstrap should also move to 18..

Comment (DovOps) [reply web-front-end/angular/package.json:1]: See my PR to your branch which will migrates ag-grid and ngx-bootstrap. Tested this all locally.

Comment (DovOps) [reply web-front-end/angular/package.json:1]: See https://github.com/mimiflynn/traderX/pull/1

**Review by DovOps — CHANGES_REQUESTED** [review]
See other comments

**Review by DovOps — APPROVED** [review]

**Comment by DovOps** [comment]
Merged via https://github.com/finos/traderX/pull/214

**Pull request closed without merging**